### PR TITLE
give template instance controller admin permissions

### LINKED ISSUE #14627: template broker can't create bindings
while provisioning the jenkins ephemeral template via the service catalog, the templateinstance object itself reports:

```
  conditions:
    - lastTransitionTime: 2017-06-14T04:58:00Z
      message: User "system:serviceaccount:openshift-infra:template-instance-controller"
        cannot create rolebindings in project "p2"
      reason: Failed
      status: "True"
      type: InstantiateFailure
```

the user has permission to create bindings in their own namespace, the template-instance-controller needs to have that permission as well.

@deads2k i assume this seems sane to you.

### PR DESCRIPTION
**Pull request by jim-minter**
fixes #14627

### CODE REVIEW

**Comment by bparees** [comment]
lgtm, @deads2k ptal as well.

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
[testextended][extended:core(template)]

File: test/extended/templates/templateinstance_security.go [hunk 1]
```diff
@@ -109,17 +109,14 @@ var _ = g.Describe("[templates] templateinstance security tests", func() {
 				},
 			},
 			{
-				// at the moment, an admin cannot create a privileged object
-				// via the template instance controller as the latter only has
-				// global edit permissions.
-				by:              "checking adminuser can't create a privileged object",
+				by:              "checking adminuser can create a privileged object",
```
Comment (bparees) [inline test/extended/templates/templateinstance_security.go]: hm.  in retrospect, we should probably have a new TC that does fail due to permissions the user has, that the controller doesn't?

File: pkg/cmd/server/bootstrappolicy/controller_policy.go [hunk 2]
```diff
@@ -133,7 +133,7 @@ func init() {
 
 	// template-instance-controller
 	controllerRoleBindings = append(controllerRoleBindings,
-		rbac.NewClusterBinding(EditRoleName).SAs(DefaultOpenShiftInfraNamespace, InfraTemplateInstanceControllerServiceAccountName).BindingOrDie())
+		rbac.NewClusterBinding(AdminRoleName).SAs(DefaultOpenShiftInfraNamespace, InfraTemplateInstanceControllerServiceAccountName).BindingOrDie())
```
Comment (deads2k) [inline pkg/cmd/server/bootstrappolicy/controller_policy.go]: I'm ok with the change.

File: test/extended/templates/templateinstance_security.go (hunk 1 above)
Comment (jim-minter) [inline test/extended/templates/templateinstance_security.go]: is something needed beyond the case immediately above this one?

Comment (bparees) [inline test/extended/templates/templateinstance_security.go]: yes.  the case above this one is "confirm we don't let the user do something the user isn't allowed to do"
the missing case is "confirm we don't do something we aren't allowed to do, even if the user is allowed to do it"

it ensures we're intersecting the two sets of permissions properly.

**Comment by bparees** [comment]
looks like some TCs need updating w/ this binding change.

**Comment by jim-minter** [comment]
@bparees repushed, ptal

**Comment by bparees** [comment]
lgtm, will merge on extended test pass.

**Comment by bparees** [comment]
[merge][severity:bug]

**Pull request merged**